source/ftrack_connect_pipeline_qt/ui/assembler/base/__init__.py

```diff
@@ -363,41 +374,51 @@ def extract_components(self, versions):
-                                        for stage in definition_fragment[key][
-                                            0
-                                        ]['stages']:
-                                            for plugin in stage['plugins']:
-                                                if not 'options' in plugin:
-                                                    plugin['options'] = {}
-                                                # Store version
-                                                plugin['options'][
-                                                    'asset_name'
-                                                ] = version['asset']['name']
-                                                plugin['options'][
-                                                    'asset_id'
-                                                ] = version['asset']['id']
-                                                plugin['options'][
-                                                    'version_number'
-                                                ] = version['version']
-                                                plugin['options'][
-                                                    'version_id'
-                                                ] = version['id']
+                                    if key != core_constants.CONTEXTS:
+                                        continue
+                                    # Remove open context
+                                    for stage in (
+                                        definition_fragment[key]
+                                        .get_first(type=core_constants.CONTEXT)
+                                        .get_all(category=core_constants.STAGE)
+                                    ):
+                                        for plugin in stage.get_all(
+                                            category=core_constants.PLUGIN
+                                        ):
+                                            if not 'options' in plugin:
+                                                plugin['options'] = {}
+                                            # Store version
+                                            plugin['options'][
+                                                'asset_name'
+                                            ] = version['asset']['name']
+                                            plugin['options'][
+                                                'asset_id'
+                                            ] = version['asset']['id']
+                                            plugin['options'][
+                                                'version_number'
+                                            ] = version['version']
+                                            plugin['options'][
+                                                'version_id'
+                                            ] = version['id']
```

Comment:
I think this can be simplyfied:
```suggestion
                                    if key != core_constants.CONTEXTS:
                                        continue
                                    # Remove open context
                                    for plugin in (
                                        definition_fragment[key]
                                        .get_all(type=core_constants.CONTEXT, category=core_constants.PLUGIN)
                                    ):
                                            if not 'options' in plugin:
                                                plugin['options'] = {}
                                            # Store version
                                            plugin['options'][
                                                'asset_name'
                                            ] = version['asset']['name']
                                            plugin['options'][
                                                'asset_id'
                                            ] = version['asset']['id']
                                            plugin['options'][
                                                'version_number'
                                            ] = version['version']
                                            plugin['options'][
                                                'version_id'
                                            ] = version['id']
```

Reply:
Well spotted, simplifying

Reply:
@henriknorin-ftrack Can't see the changes reflected.